src/printer.js

```diff
@@ -2843,7 +2843,11 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
     // the other ones since we don't call the normal print on BinaryExpression,
     // only for the left and right parts
     if (isNested && node.comments) {
-      parts.push(comments.printComments(path, p => "", options));
+      parts.splice(
+        0,
+        parts.length,
+        comments.printComments(path, p => concat(parts.slice()), options)
+      );
```

Comment:
Ah, yep. I remember now. I think I tried to avoid this mutation at some point but it was pretty hard with the recursive call, and I want to land a working implementation. The recursiveness makes it pretty hard, and the resulting code might be a bit more awkward. I forget the details, but feel free to give it a try.

Worst case we can merge this. It's not the best, but oh well..